TestFunctional/parallel/DryRun: exit code = -1, wanted = 78: signal: killed

I believe the dry run returns a different exist code on kvm for some reason !
as seen in https://storage.googleapis.com/minikube-builds/logs/6338/KVM_Linux.html#failtestwindow5


```
=== RUN   TestFunctional/parallel/DryRun
=== PAUSE TestFunctional/parallel/DryRun
=== CONT  TestFunctional/parallel/DryRun
--- FAIL: TestFunctional/parallel/DryRun (3.07s)
functional_test.go:313: (dbg) Run:  out/minikube-linux-amd64 start -p functional-20200121T174445.238510036-4853 --dry-run --memory 250MB --vm-driver=kvm2 
functional_test.go:313: (dbg) Non-zero exit: out/minikube-linux-amd64 start -p functional-20200121T174445.238510036-4853 --dry-run --memory 250MB --vm-driver=kvm2 : signal: killed (2.001806296s)
-- stdout --
* [functional-20200121T174445.238510036-4853] minikube v1.7.0-beta.0 on Debian 9.11
- KUBECONFIG=/home/jenkins/minikube-integration/linux-amd64-kvm2-6338-2439-5a3c13815ed5bb198a92b2183d9542af4462ddb3/kubeconfig
- MINIKUBE_BIN=out/minikube-linux-amd64
- MINIKUBE_HOME=/home/jenkins/minikube-integration/linux-amd64-kvm2-6338-2439-5a3c13815ed5bb198a92b2183d9542af4462ddb3/.minikube
- MINIKUBE_LOCATION=6338
-- /stdout --
functional_test.go:317: dry-run(250MB) exit code = -1, wanted = 78: signal: killed
functional_test.go:324: (dbg) Run:  out/minikube-linux-amd64 start -p functional-20200121T174445.238510036-4853 --dry-run --vm-driver=kvm2 
functional_test.go:324: (dbg) Done: out/minikube-linux-amd64 start -p functional-20200121T174445.238510036-4853 --dry-run --vm-driver=kvm2 : (1.071343365s)
```

but not seen in same test for vbox https://storage.googleapis.com/minikube-builds/logs/6338/VirtualBox_Linux.html

https://storage.googleapis.com/minikube-builds/logs/7389/eb42d16/KVM_Linux.html

```
--- FAIL: TestFunctional/parallel/DryRun (11.55s)
functional_test.go:387: (dbg) Run:  out/minikube-linux-amd64 start -p functional-20200406T171223.163798203-5853 --dry-run --memory 250MB --alsologtostderr -v=1 --driver=kvm2 
functional_test.go:387: (dbg) Non-zero exit: out/minikube-linux-amd64 start -p functional-20200406T171223.163798203-5853 --dry-run --memory 250MB --alsologtostderr -v=1 --driver=kvm2 : signal: killed (5.125049473s)
** stderr ** 
	I0406 17:22:00.826966   13399 notify.go:125] Checking for updates...
	I0406 17:22:02.464601   13399 start.go:262] hostinfo: {"hostname":"kvm-integration-slave6","uptime":3863,"bootTime":1586215059,"procs":269,"os":"linux","platform":"debian","platformFamily":"debian","platformVersion":"9.12","kernelVersion":"4.9.0-12-amd64","virtualizationSystem":"kvm","virtualizationRole":"host","hostid":"ae41e7f6-8b8e-4d40-b77d-1ebb5a2d5fdb"}
	I0406 17:22:02.465621   13399 start.go:272] virtualization: kvm host
** /stderr **
functional_test.go:391: dry-run(250MB) exit code = -1, wanted = 78: signal: killed
functional_test.go:398: (dbg) Run:  out/minikube-linux-amd64 start -p functional-20200406T171223.163798203-5853 --dry-run --alsologtostderr -v=1 --driver=kvm2 
functional_test.go:398: (dbg) Non-zero exit: out/minikube-linux-amd64 start -p functional-20200406T171223.163798203-5853 --dry-run --alsologtostderr -v=1 --driver=kvm2 : signal: killed (5.80576278s)
** stderr ** 
	I0406 17:22:06.747915   13413 notify.go:125] Checking for updates...
** /stderr **
functional_test.go:400: dry-run exit code = -1, wanted = 0: signal: killed
```

The root cause appears to be that we are checking for updates in dry-run mode. Sometimes that check takes a while.

--- a/cmd/minikube/cmd/start.go
+++ b/cmd/minikube/cmd/start.go
@@ -286,6 +286,13 @@ func platform() string {
 // runStart handles the executes the flow of "minikube start"
 func runStart(cmd *cobra.Command, args []string) {
 	displayVersion(version.GetVersion())
+
+	// No need to do the update check if no one is going to see it
+	if !viper.GetBool(interactive) || !viper.GetBool(dryRun) {
+		// Avoid blocking execution on optional HTTP fetches
+		go notify.MaybePrintUpdateTextFromGithub()
+	}
+
 	displayEnviron(os.Environ())
 
 	// if --registry-mirror specified when run minikube start,
@@ -406,12 +413,7 @@ func displayVersion(version string) {
 		prefix = fmt.Sprintf("[%s] ", ClusterFlagValue())
 	}
 
-	versionState := out.Happy
-	if notify.MaybePrintUpdateTextFromGithub() {
-		versionState = out.Meh
-	}
-
-	out.T(versionState, "{{.prefix}}minikube {{.version}} on {{.platform}}", out.V{"prefix": prefix, "version": version, "platform": platform()})
+	out.T(out.Happy, "{{.prefix}}minikube {{.version}} on {{.platform}}", out.V{"prefix": prefix, "version": version, "platform": platform()})
 }
 
 // displayEnviron makes the user aware of environment variables that will affect how minikube operates

